### PR TITLE
WIP: having prometheus as library 

### PR DESCRIPTION
**Pull request by itsmylife**
**What is this feature?**

_TBA_

### CODE REVIEW

File: pkg/tsdb/prometheus/go.mod [hunk 1]
```diff
@@ -0,0 +1,136 @@
+module github.com/grafana/grafana/pkg/tsdb/prometheus
```
Comment (kylebrandt) [inline pkg/tsdb/prometheus/go.mod:1]: I don't think the whole DS will become a module? I imagine there is some stub in tsdb/prometheus that references most of the code (which maybe lives on level deeper?) that is the "core/vanilla prom ds"... that make sense?

**Comment by bohandley** [comment]
Is this exporting everything we have in the Prometheus package? Are we exporting azure or aws code?

File: pkg/tsdb/prometheus/go.mod (hunk 1 above)
Comment (itsmylife) [reply pkg/tsdb/prometheus/go.mod:1]: This is just a test to see how CI is reacting. We will close this when we are done with testing.

**Comment by itsmylife** [comment]
> Is this exporting everything we have in the Prometheus package? Are we exporting azure or aws code?

Please see https://github.com/grafana/grafana/pull/82905#discussion_r1493367860

**Comment by itsmylife** [comment]
closing this to test the idea with a simpler and harmless package.

**Pull request closed without merging**